net/procfs/net_procfs.c

```diff
@@ -283,41 +328,14 @@ static ssize_t netprocfs_read(FAR struct file *filep, FAR char *buffer,
 
   /* Read according to the sub-directory */
 
-  switch (priv->entry)
+  for (i = 0; i < ARRAY_SIZE(g_net_entries); i++)
```

Comment:
why not save array index directly to avoid search again and remove netprocfs_entry_e?

Reply:
Done